### PR TITLE
Add async await syntax.

### PR DESCRIPTION
**Pull request by Exitare**
Add the possibility to use async/await as discussed in  #56
E.g. 

`    
const Hook: webhook.Webhook = new webhook.Webhook("WEBHOOK URL");
          
const msg = new webhook.MessageBuilder().setAuthor("TEST_AUTHOR");      
      
await Hook.send(msg);
`

### CODE REVIEW

File: lib/webhook.js [hunk 1]
```diff
@@ -302,12 +302,12 @@ class Webhook {
             if (
                 messageBuilder.data.text // Check for any text
             ) {
-                return this.sendPayload(messageBuilder.data, resolve);
+                return await this.sendPayload(messageBuilder.data, resolve);
```
Comment (jos-b) [inline lib/webhook.js:305]: Do we still need a resolve function? This was messy when I wrote it and still is.

**Comment by jos-b** [comment]
Accidentally marked as approved

**Comment by jos-b** [comment]
Also looks like there is a gnarly eslint error in there

**Comment by Exitare** [comment]
The benefit is to use async and await syntax. As discussed. This pr is just to introduce them and make it available. It does not fix bad coding decisions nor any logical errors. This should be handled by other pull requests.

File: lib/webhook.js (hunk 1 above)
Comment (Exitare) [reply lib/webhook.js:305]: Then this should be resolved in future commits. Not in this pr.

**Comment by jos-b** [comment]
@Exitare the implementation you have proposed is broken. It does not work at all, I appreciate another PR should fix "poor coding decisions" but this is not working by any means.

```
/Users/joseph/Desktop/Projects/JavaScript/Node.js/webhook-discord/lib/webhook.js:109
                                    await this.sendPayload(payload, resolve);
                                    ^^^^^

SyntaxError: await is only valid in async function
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:657:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
```

File: lib/webhook.js (hunk 1 above)
Comment (jos-b) [reply lib/webhook.js:305]: No, this resolve function was introduced with the old async system, it's just unused now.

If we introduce async/await syntax then we go all the way, we don't mess around with old style promises and new style.

**Comment by Exitare** [comment]
That wasn’t meant to be rude. Anyways, I don’t have any time to spare in the foreseeable future and therefore will close this PR. If someone would like to take over feel free to do that.

**Comment by jos-b** [comment]
I'll look into it. I do agree though that the current async system is iffy, if it is reimplemented it should be a full rewrite of all the promises.

**Pull request closed without merging**